### DISCUSSION TITLE
Dotmailer yes/no field doesn't map

### DISCUSSION THREAD

**Discussion by LoriWJ** [solspace/craft-freeform#636] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**
I'm using the Dotmailer integration with the form. Dotmailer is now rebranded as DotDigital. In the field settings, I can map the form fields to the data fields in Dotmailer/DotDigital. This works fine, except when I map a radio group form field with a yes/no Dotmailer/DotDigital field. It is recorded as "yes" in Dotmailer/DotDigital even if you select "no" in the radio group form field.

**Steps to reproduce**

1. In Dotmailer/DotDigital, add a yes/no data field. 
2. Set up a Freeform form with a radio group form field.
3. For the values on the radio group form field, use "yes" and "no".
4. Add the Dotmailer integration form field.
5. In the Dotmailer integration form field settings, map the yes/no data field with the radio group form field.

**Expected behavior**
If you select "no" in the radio group form field, the yes/no data field should record "no".

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 4.3.4
- Freeform Version: 4.0.8
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Additional context**

**Comment by kjmartens** [comment]
Sorry for the trouble and the delay @LoriWJ. I'll have a developer check into this and get back to you soon. 🙂

**Comment by kjmartens** [comment, marked as answer]
Hi @LoriWJ,

I checked with a developer and it turns out this is behaving as expected.

The `yes`/`no` field in _dotmailer_ is a boolean field which looks like `yes` and `no` when behind the scenes it is actually `true` and `false`. When using a Freeform radio field with `yes` and `no` values (which are strings), when the form gets posted, both `yes` and `no` strings get converted to `true` since they’re not empty strings.

A possible solution would be to assign the `no` radio value as an **empty** string instead. 🙂
